File: backend/backend/src/SignalingComponent/Common/DTOs/SDPMessage.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using backend.WebSocketComponent.Common.DTOs;

namespace backend.SignalingComponent.Common.DTOs;

public class SdpMessage : TypedMessage
{
    public override string GetTypeString()
    {
        return "sdp-message";
    }
    
    [JsonPropertyName("remoteToken")]
    
    public required string RemoteToken { get; set; }
    
    [JsonPropertyName("description")]
    
    public JsonElement Description { get; set; }
}

File: frontend/src/types/rtc/SdpMessage.ts
import { ITypedMessage } from "../ITypedMessage";
import { MessageType } from "../MessageType";

export class SdpMessage implements ITypedMessage {
    public readonly type = MessageType.SDP;
    public msg: {
        remoteToken: string;
        description: RTCSessionDescription;
    };

    public constructor(msg: {
        remoteToken: string;
        description: RTCSessionDescription;
    }) {
        this.msg = msg;
    }
}
